google-cloud-spanner/lib/google/cloud/spanner/service.rb

```diff
@@ -376,12 +389,10 @@ def inspect
 
         protected
 
-        def default_headers
-          { "google-cloud-resource-prefix" => "projects/#{@project}" }
-        end
-
-        def default_options
-          Google::Gax::CallOptions.new kwargs: default_headers
+        def default_options_from_session session_name
+          default_prefix = session_name.split("/sessions/").first
```

Comment:
The value of `default_prefix` will be the most specific of database name, instance name, or project name, correct?